components/SearchAutocomplete.tsx

```diff
@@ -24,35 +24,32 @@ import {
 import useOutsideClickHandler from 'hooks/useOutsideClickHandler'
 import Group, { GroupProps } from './Group'
 
-export interface SearchOptionProps {
+export interface SearchOptionGroupType<OptionType> {
   label: string
-  value: string
+  options: OptionType[]
 }
 
-export interface SearchOptionGroupType<T extends SearchOptionProps> {
-  label: string
-  options: T[]
-}
-
-interface SearchAutocompleteProps<OptionType extends SearchOptionProps> {
+interface SearchAutocompleteProps<OptionType> {
   value?: OptionType | null
   InputProps?: IProps
   options?: (OptionType | SearchOptionGroupType<OptionType>)[]
   emptyOption?: ReactNode
   renderOption?: (option: OptionType) => ReactNode
   onSelectOption?: (option: OptionType | null) => void
+  getOptionLabel: (option: OptionType | null) => string
   inputLeftElement?: ReactNode
   groupProps?: GroupProps
   variant?: string
 }
 
-const SearchAutocomplete = <OptionType extends SearchOptionProps>({
+const SearchAutocomplete = <OptionType,>({
```

Comment:
Is this comma intentional / valid?

Reply:
Yes as I read its added due to  contraints of the .tsx file extension